ci_cd/utils/versions.py

```diff
@@ -292,7 +367,7 @@ def next_version(self, version_part: str) -> SemanticVersion:
         return self.__class__(next_version)
 
     def previous_version(
-        self, version_part: str, max_filler: str | int | None = 99
+        self, version_part: str, max_filler: str | int | None = None
```

Comment:
The logic for determining the previous version seems overly complex and could be simplified. For instance, using a dictionary to map version parts to their decrement logic might make the code more readable and maintainable.

Reply:
Maybe. But this way, the decrement logic will only ever run for the given version_part. If a dictionary/mapping is created, the whole of its contents will have to be resolved before then choosing which one is needed.

Sometimes readability is valued higher than efficiency and optimized run times, however, the change in readability is minimal here, I'd argue.

It's basically already split into two parts:
1. Value and Type checks for inputs / set defaults.
2. If-branching depending on the three available `version_part` values.

It's the same for `next_version()`, however, since there is no `max_filler` variable to worry about, this part can be left out. Also, the logic for determining a previous version for some version parts are a bit more involved.